core/server/master/src/main/java/alluxio/master/backup/BackupWorkerRole.java

```diff
@@ -215,6 +220,11 @@ private void handleRequestMessage(BackupRequestMessage requestMsg) {
           enforceResumeJournals();
         } else {
           LOG.info("Journal transition completed. Scheduling a backup.");
+          if (!mBackupTimeoutTask.cancel(true)) {
+            mBackupTracker
+                .updateError(new BackupException("Journal has ben resumed due to time-out."));
```

Comment:
When the advance future being cancelled, journal will be automatically resume?

Reply:
It should.
However it looks like CompletableFuture<?> does not support cancelling. So I'll need to change it in the signature to Future<?> and re-implement related parts.

Reply:
@LuQQiu  `CompletableFuture<?>` was not cancelleable. 

`JournalSystem#advance` now returns `AdvanceFuture` which allows for graceful termination of ongoing advancing.